src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.FormatInfo.cs

```diff
@@ -50,8 +50,65 @@ internal static int GetFormatInfoForMountPoint(string name, out DriveType type)
             return GetFormatInfoForMountPoint(name, out _, out type);
         }
 
+        /// <summary>
+        /// Retrieves format information for the specified mount point.
+        /// </summary>
+        /// <remarks>
+        /// This method uses a two-step approach to retrieve filesystem information:
+        /// 1. On Linux systems, it first attempts to read from `/proc/self/mountinfo`.
+        /// 2. If step 1 fails or on non-Linux systems, it falls back to a P/Invoke call.
+        ///
+        /// The `/proc/self/mountinfo` approach is preferred on Linux because
+        /// it's more reliable when procfs is available and functioning correctly.
+        ///
+        /// For other systems:
+        /// - SunOS and similar: procfs provides a binary interface, not suitable for this method.
+        /// - macOS: procfs is not available.
+        /// - FreeBSD: procfs is optional and not enabled by default.
+        ///
+        /// The method uses try/catch blocks to ensure robustness, even though `/proc/self/mountinfo`
+        /// should be specific to each process according to kernel documentation.
+        /// </remarks>
+        /// <param name="name">The mount point name to query.</param>
+        /// <param name="format">Output parameter for the filesystem format.</param>
+        /// <param name="type">Output parameter for the drive type.</param>
+        /// <returns>0 if successful, otherwise an error code.</returns>
         private static unsafe int GetFormatInfoForMountPoint(string name, out string format, out DriveType type)
         {
+#if TARGET_LINUX
+            try
+            {
+                using StreamReader reader = new("/proc/self/mountinfo");
+
+                string rawLine;
+                while ((rawLine = reader.ReadLine()) is not null)
+                {
+                    ReadOnlySpan<char> line = rawLine.AsSpan();
+                    MemoryExtensions.SpanSplitEnumerator<char> fields = line.Split(' ');
+
+                    // Skip fields we don't care about (Fields 1-4)
+                    fields.MoveNext(); // Skip Mount ID
+                    fields.MoveNext(); // Skip Parent ID
+                    fields.MoveNext(); // Skip Major:Minor
+                    fields.MoveNext(); // Skip Root
+                    fields.MoveNext(); // Skip to MountPoint field
+
+                    if (!MemoryExtensions.Equals(line[fields.Current], name, StringComparison.Ordinal)) continue;
+
+                    // Skip to the separator which is end of optional fields (Field 8)
+                    while (fields.MoveNext() && !MemoryExtensions.Equals(line[fields.Current], "-", StringComparison.Ordinal));
+
+                    fields.MoveNext();
+                    format = line[fields.Current].ToString();
+
+                    fields.MoveNext();
+                    type = GetDriveType(line[fields.Current].ToString());
+
+                    return 0;
+            }
+            catch { /* ignored */ }
```

Comment:
I see this was discussed in https://github.com/dotnet/runtime/pull/107365#issuecomment-2330774825.

Personally, I think the `File.Exists` should be fine, but I defer to the maintainers' preferences.

Reply:
The reason why all/most procfs usage in libs are using try-catch pattern is because anything could go wrong (https://github.com/dotnet/runtime/issues/62513). Also see https://github.com/dotnet/runtime/issues/74316#issuecomment-1221685984 and https://github.com/dotnet/runtime/issues/52753 to read on unreliability issues FileSystemInfo.Exists out in the wild.

Reply:
The reason why all/most procfs usage in libs are using try-catch pattern is because anything could go wrong (https://github.com/dotnet/runtime/issues/62513). Also see https://github.com/dotnet/runtime/issues/74316#issuecomment-1221685984 and https://github.com/dotnet/runtime/issues/52753 to read on unreliability issues attached with FileSystemInfo.Exists out in the wild.

Reply:
There are some reasons why we don't expect these to occur for `/proc/self/mountinfo`

Anyway, my main concern here is that a generic catch all masks failures in the parsing.

Perhaps we can filter for IOException/UnauthorizedAccessException. These are the exceptions we might expect when not being able to read the procfs file.